lib/gpgme/key.rb

```diff
@@ -122,6 +122,12 @@ def import(keydata, options = {})
           ctx.import_result
         end
       end
+
+      # Checks if a key is valid
+      def valid?(key)
+        GPGME::Key.import(key).considered == 1
+      end
```

Comment:
@alexjfisher 

one idea could be to check if that key is already imported. 
if not, we deleted it afterwards again.
it's a bit ugly, but at least the keyring stays clean. would that be a sufficient effort?